client/src/app/site/services/view-model-store-update.service.ts

```diff
@@ -58,24 +58,38 @@ export class ViewModelStoreUpdateService {
         }
 
         for (const collection of Object.keys(patch)) {
-            _changedModels[collection] = this.createCollectionUpdate(collection, patch);
+            const { update, toDelete } = this.createCollectionUpdate(collection, patch);
+            _changedModels[collection] = update;
+            _deletedModels[collection] = (_deletedModels[collection] || []).concat(toDelete);
         }
 
         await this.doCollectionUpdates(_changedModels, _deletedModels);
     }
 
-    private createCollectionUpdate(collection: Collection, modelData: ModelPatch): BaseModel<any>[] {
+    private createCollectionUpdate(
+        collection: Collection,
+        modelData: ModelPatch
+    ): { update: BaseModel<any>[]; toDelete: number[] } {
         const update: BaseModel<any>[] = [];
+        const toDelete: number[] = [];
         for (const id of Object.keys(modelData[collection])) {
             const model = modelData[collection][+id];
-            // Important: our model system needs to have an id in the model, even if it is partial
-            model[`id`] = +id;
-            const basemodel = this.mapObjectToBaseModel(collection, model);
-            if (basemodel) {
-                update.push(basemodel);
+            if (this.isEmpty(model)) {
```

Comment:
Better check only for the id. If the id is null, then the model was deleted, if not then not. All other cases cannot happen and are prevented by the autoupdate service.